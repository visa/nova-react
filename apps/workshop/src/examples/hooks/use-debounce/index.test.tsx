/**
 *              Copyright (c) 2025 Visa, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 **/
import { act, fireEvent, render, screen } from '@testing-library/react';

import metaData from './meta.json';

import { UseButtonDebounceExample } from './use-button-debounce-example';

const examples = [{ Component: UseButtonDebounceExample, title: metaData['use-button-debounce-example'].title }];

beforeAll(() => jest.useFakeTimers());
afterAll(() => jest.useRealTimers());

describe('useDebounce example', () => {
  examples.forEach(({ Component, title }) => {
    it(`${title} should render correctly`, async () => {
      const { container } = render(<Component />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('button debounce', () => {
    it('should show success state after debounce and reset', () => {
      render(<UseButtonDebounceExample />);

      const submitButton = screen.getByText('Submit debounced');
      fireEvent.click(submitButton);

      // Advance timers by the delay
      act(() => {
        jest.advanceTimersByTime(1000);
      });

      expect(screen.queryByText('Button click successful, many thanks')).toBeInTheDocument();

      const resetButton = screen.getByText('Reset');
      fireEvent.click(resetButton);

      expect(screen.queryByText('Waiting for button click')).toBeInTheDocument();
    });
  });
});
