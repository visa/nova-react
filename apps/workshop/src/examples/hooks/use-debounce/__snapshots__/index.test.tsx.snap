// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useDebounce example Behavior and usage should render correctly 1`] = `
<div>
  <div
    class="v-flex v-flex-col v-gap-12"
  >
    <span
      class="v-typography-body-2-bold"
    >
      Rapidly click the button in sequence, then wait for one second:
    </span>
    <div
      class="v-flex v-flex-row v-gap-12"
    >
      <button
        class="v-button"
      >
        Submit debounced
      </button>
      <button
        class="v-button v-button-secondary"
      >
        Reset
      </button>
    </div>
    <p
      aria-atomic="true"
      aria-live="assertive"
      style="line-height: 16px;"
    >
      <svg
        aria-hidden="true"
        class="v-icon v-icon-visa v-icon-tiny v-icon-information"
        focusable="false"
        height="16"
        style="--v-icon-primary: var(--palette-default-text-subtle); --v-icon-secondary: var(--palette-default-text-subtle); margin-inline-end: 3px; vertical-align: bottom;"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="v-icon-primary"
          clip-rule="evenodd"
          d="M8 16C12.4183 16 16 12.4183 16 8C16 3.58172 12.4183 0 8 0C3.58172 0 0 3.58172 0 8C0 12.4183 3.58172 16 8 16ZM7 5V3H9V5H7ZM6 9H7V10V11H6V13H7H9H10V11H9V10V8V7H6V9Z"
          fill-rule="evenodd"
        />
        <defs>
          <clippath>
            <rect
              height="16"
              width="16"
            />
          </clippath>
        </defs>
      </svg>
      Waiting for button click
    </p>
  </div>
</div>
`;
